### PR TITLE
Add I2cTransferResult and I2cTransferStatus

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add I2cTransferResult and I2cTransferStatus.
- Changed this into a "core libray".

## Motivation and Context
- These are required to provide a "developer friendly" feedback regarding the I2C bus transaction. 

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: System.Device.I2c/I2cDevice.cs [hunk 1]
```diff
@@ -11,25 +11,40 @@ namespace System.Device.I2c
...
-        // For the ReadByte and WriteByte operations
-        private byte[] bufferSingleOperation = new byte[1];
+        [Diagnostics.DebuggerBrowsable(Diagnostics.DebuggerBrowsableState.Never)]
+        private readonly I2cConnectionSettings _connectionSettings;
+
+        [Diagnostics.DebuggerBrowsable(Diagnostics.DebuggerBrowsableState.Never)]
+        private bool _disposed;
+
+        // speeds up the execution of ReadByte and WriteByte operations
+        private readonly byte[] _buffer;
```
Comment (Ellerbach) [inline System.Device.I2c/I2cDevice.cs:26]: This now should be SpanByte

File: System.Device.I2c/I2cDevice.cs [hunk 2]
```diff
@@ -11,25 +11,40 @@ namespace System.Device.I2c
...
         /// Reads a byte from the I2C device.
         /// </summary>
         /// <returns>A byte read from the I2C device.</returns>
         public byte ReadByte()
         {
-            _device.Read(bufferSingleOperation);
-            return bufferSingleOperation[0];
+            lock (_syncLock)
+            {
+                var buffer = new SpanByte(_buffer);
```
Comment (Ellerbach) [inline System.Device.I2c/I2cDevice.cs:42]: this is not needed as _buffer should be a SpanByte directly

File: System.Device.I2c/I2cDevice.cs [hunk 3]
```diff
@@ -11,25 +11,40 @@ namespace System.Device.I2c
...
+
+                NativeTransmit(null, buffer);
```
Comment (Ellerbach) [inline System.Device.I2c/I2cDevice.cs:44]: And here it's directly _buffer

File: System.Device.I2c/I2cDevice.cs [hunk 4]
```diff
@@ -39,26 +54,27 @@ public byte ReadByte()
...
+        public I2cTransferResult WriteByte(byte value)
         {
-            bufferSingleOperation[0] = value;
-            _device.Write(bufferSingleOperation);
+            lock (_syncLock)
+            {
+                // copy value
+                _buffer[0] = value;
+
+                return NativeTransmit(new SpanByte(_buffer), null);
```
Comment (Ellerbach) [inline System.Device.I2c/I2cDevice.cs:76]: not needed, _buffer should now be a SpanByte

File: System.Device.I2c/I2cDevice.cs [hunk 5]
```diff
@@ -114,26 +127,65 @@ public static I2cDevice Create(I2cConnectionSettings settings)
         /// <param name="settings">Connection settings</param>
         public I2cDevice(I2cConnectionSettings settings)
         {
-            ConnectionSettings = settings;
-            _device = Windows.Devices.I2c.I2cDevice.FromId($"I2C{settings.BusId}", new Windows.Devices.I2c.I2cConnectionSettings(settings.DeviceAddress)
+            _connectionSettings = settings;
+
+            // create the buffer
+            _buffer = new byte[1];
```
Comment (Ellerbach) [inline System.Device.I2c/I2cDevice.cs:133]: This is ok, it's just than _buffer should be a SpanByte, this new allocation won't change

**Review by Ellerbach — COMMENTED** [review]
Looks great, just _buffer which should be a SpanByte

File: System.Device.I2c/I2cDevice.cs (hunk 2 above)
Comment (josesimoes) [reply System.Device.I2c/I2cDevice.cs:42]: Can't do...
CS8345	Field or auto-implemented property cannot be of type 'SpanByte' unless it is an instance member of a ref struct.

File: System.Device.I2c/I2cDevice.cs (hunk 1 above)
Comment (josesimoes) [reply System.Device.I2c/I2cDevice.cs:26]: Can't do...
CS8345 Field or auto-implemented property cannot be of type 'SpanByte' unless it is an instance member of a ref struct.

Comment (Ellerbach) [reply System.Device.I2c/I2cDevice.cs:26]: hum, then it makes less sense to do this. We can tweak SpanByte to be a ref struct (it's technically one on the Span<byte> world.

**Review by Ellerbach — APPROVED** [review]
Understand why can't be SpanByte for _buffer, then all good like this

**Pull request merged**